designer/client/src/components/modals/ValidationLabels.tsx

```diff
@@ -21,35 +23,20 @@ const LabelsContainer = styled("div")({
 });
 
 export default function ValidationLabels(props: Props) {
-    const { validators, values, validationLabelInfo } = props;
-
-    const validationErrors: ValidationError[] = withoutDuplications(validators)
-        .filter((v) => !v.isValid(...values))
-        .map((validator) => ({
-            message: validator.message && validator.message(),
-            description: validator.description && validator.description(),
-        }));
-
-    const isValid: boolean = isEmpty(validationErrors);
-
-    const renderErrorLabels = () =>
-        validationErrors.map((validationError) => {
-            // we don't pass description as tooltip message until we make changes on the backend
-            return (
-                <LimitedValidationLabel key={validationError.message} title={validationError.message} type="ERROR">
-                    {validationError.message}
-                </LimitedValidationLabel>
-            );
-        });
+    const { fieldErrors, validationLabelInfo } = props;
 
     // TODO: We're assuming that we have disjoint union of type info & validation errors, which is not always the case.
     // It's possible that expression is valid and it's type is known, but a different type is expected.
     return (
         <LabelsContainer>
-            {isValid ? (
+            {isEmpty(fieldErrors) ? (
                 <LimitedValidationLabel title={validationLabelInfo}>{validationLabelInfo}</LimitedValidationLabel>
             ) : (
-                renderErrorLabels()
+                fieldErrors.map((fieldErrors) => (
+                    <LimitedValidationLabel key={fieldErrors.message} title={fieldErrors.message} type="ERROR">
```

Comment:
Now that we are display all errors coming from BE, I don't think we can use message as key. When typing an expression that returns two exactly same errors like `a + a` (without having a in context), then removing `a`, then adding it again we get 3 errors. React throws `Uncaught Error: Children cannot be added or removed during a reorder operation.`

Can we generate ids on FE without messing up ordering of errors?

Reply:
Nice catch! I didn't expect the messages to be the same, and I ignored this key (I just copied a preview approach). I just changed the key to the index, which resolves the uniqueness issue.